ironfish-cli/src/commands/accounts/rescan.ts

```diff
@@ -22,34 +22,56 @@ export class RescanCommand extends IronfishCommand {
       description:
         'clear the in-memory and disk caches before rescanning. note that this removes all pending transactions',
     }),
+    offline: flags.boolean({
+      default: false,
+      description: 'Rescan the blockchain without an online node',
+    }),
   }
 
   async start(): Promise<void> {
     const { flags } = this.parse(RescanCommand)
+    const { follow, reset, offline } = flags
 
-    await this.sdk.client.connect()
-    await rescan(this.sdk.client, flags.follow, flags.reset)
+    await rescan(this.sdk, follow, reset, offline)
   }
 }
 
 export async function rescan(
-  client: IronfishRpcClient,
+  sdk: IronfishSdk,
   follow: boolean,
   reset: boolean,
+  offline: boolean,
 ): Promise<void> {
   cli.action.start('Rescanning Transactions', 'Asking node to start scanning', {
     stdout: true,
   })
 
-  const startedAt = Date.now()
-  const response = client.rescanAccountStream({ follow, reset })
-
   try {
-    for await (const result of response.contentStream()) {
-      cli.action.status = `Scanning Block: ${result.sequence}, ${Math.floor(
+    const updateCliStatus = (startedAt: number) => ({
+      sequence,
+    }: {
+      sequence: number
+    }): void => {
+      cli.action.status = `Scanning Block: ${sequence}, ${Math.floor(
         (Date.now() - startedAt) / 1000,
       )} seconds`
     }
+    const updateCliStatusWithStartTime = updateCliStatus(Date.now())
+
+    if (offline) {
```

Comment:
We can reduce a lot of this code to just use `getConnectedClient` from `config:show` which at some point needs to be expanded out.

Reply:
Ah, I hadn't seen that before, that's convenient.